jxlint-gradle/src/main/java/com/selesse/jxlint/gradle/AbstractJxlintTask.java

```diff
@@ -0,0 +1,242 @@
+
+import java.io.File;
+import java.util.List;
+import java.util.function.Consumer;
+
+public abstract class AbstractJxlintTask extends DefaultTask {
+
+    public static final String DISABLE_RULES = "disableRules";
+    public static final String ENABLE_ONLY_RULES = "enableOnlyRules";
+    public static final String ENABLE_RULES = "enableRules";
+    public static final String ENABLE_CATEGORY = "enableCategories";
+    public static final String SOURCE_DIRECTORY = "sourceDirectory";
+    public static final String NO_WARNINGS = "noWarnings";
+    public static final String ALL_WARNINGS = "allWarnings";
+    public static final String WARNINGS_ARE_ERRORS = "waringsAreErrors";
+    public static final String OUTPUT_TYPE = "outputType";
+    public static final String OUTPUT_FILE = "outputFile";
+    public static final String SRC_PATH_PREFIX = "srcPathPrefix";
+
+    protected List<String> disableRules;
+
+    protected List<String> enableOnlyRules;
+
+    protected List<String> enableRules;
+
+    protected List<String> enableCategories;
+
+    protected String sourceDirectory;
+
+    protected boolean noWarnings = false;
+
+    protected boolean allWarnings = false;
+
+    protected boolean waringsAreErrors = false;
+
+    protected String outputType = "html";
+
+    protected String outputFile;
+
+    protected String srcPathPrefix;
+
+    @Option(option = DISABLE_RULES, description = "Disable the list of rules.")
+    public void setDisableRules(List<String> disableRules) {
+        this.disableRules = disableRules;
+    }
+
+    @Option(option = ENABLE_ONLY_RULES, description = "Only check for these rules.")
+    public void setEnableOnlyRules(List<String> enableOnlyRules) {
+        this.enableOnlyRules = enableOnlyRules;
+    }
+
+    @Option(option = ENABLE_RULES, description = "Enable the list of rules.")
+    public void setEnableRules(List<String> enableRules) {
+        this.enableRules = enableRules;
+    }
+
+    @Option(option = ENABLE_CATEGORY, description = "Run all rules of a certain category.")
+    public void setEnableCategories(List<String> enableCategories) {
+        this.enableCategories = enableCategories;
+    }
+
+    @Option(option = SOURCE_DIRECTORY, description = "Directory where the sources to analyze are located")
+    public void setSourceDirectory(String sourceDirectory) {
+        this.sourceDirectory = sourceDirectory;
+    }
+
+    @Option(option = NO_WARNINGS, description = "Only check for errors; ignore warnings.")
+    public void setNoWarnings(boolean noWarnings) {
+        this.noWarnings = noWarnings;
+    }
+
+    @Option(option = ALL_WARNINGS, description = "Check all warnings, including those off by default.")
+    public void setAllWarnings(boolean allWarnings) {
+        this.allWarnings = allWarnings;
+    }
+
+    @Option(option = WARNINGS_ARE_ERRORS, description = "Treat all warnings as errors.")
+    public void setWaringsAreErrors(boolean waringsAreErrors) {
```

Comment:
```suggestion
    public void setWarningsAreErrors(boolean warningsAreErrors) {
```

Reply:
Thank you for this review!

Reply:
Same change for the maven plugin: https://github.com/selesse/jxlint/pull/41